readthedocs/core/views/hooks.py

```diff
@@ -48,7 +48,13 @@ def build_branches(project, branch_list):
     to_build = set()
     not_building = set()
     for branch in branch_list:
-        versions = project.versions_from_branch_name(branch)
+        # Used for External Versions
+        # pull/merge request `handle_webhook()` sends version instance for Building.
```

Comment:
Hrm, I wonder if we could make this fit the existing logic, or change how we pass in versions other places here. I guess this is fine though.

Reply:
We would have to do a lot of refactor to change this. I think its a good approach for branches or tags as we get the `ref`. but in the case of PR Versions we create the versions first and the verbose name could match the other branch or tag name.

Reply:
We can avoid this by calling another method (instead of `get_response_push`) in `get_external_version_response`
And even change the response to something more related to the PR like with `sync_versions` https://github.com/rtfd/readthedocs.org/pull/5823/files#diff-23bf1ad655dc56b6bd5a9f7958b52ebbR291